language/diem-framework/modules/0L/MakeWhole.move

```diff
@@ -20,139 +22,102 @@ address 0x1 {
         const EALREADY_PAID: u64 = 22017;
 
 
-        public fun make_whole_init(vm: &signer){
+        public fun vm_offer_credit(
```

Comment:
Could I get an ELI5 for how this function is being run with both a vm signer and an account signer? Is it being called in a block prologue, or through some other mechanism?